packages/flutter_tools/lib/src/macos/xcode.dart

```diff
@@ -185,3 +189,327 @@ class Xcode {
+    } on ArgumentError catch (exception) {
+      _logger.printTrace('Argument exception running xcdevice list:\n$exception');
+    }
+
+    return null;
+  }
+
+  List<dynamic> _cachedListResults;
+
+  /// List of devices available over USB.
+  Future<List<IOSDevice>> getAvailableTetheredIOSDevices() async {
+    final List<dynamic> allAvailableDevices = await _getAllDevices();
+
+    if (allAvailableDevices == null) {
+      return const <IOSDevice>[];
+    }
+
+    // [
+    //  {
+    //    "simulator" : true,
+    //    "operatingSystemVersion" : "13.3 (17K446)",
+    //    "available" : true,
+    //    "platform" : "com.apple.platform.appletvsimulator",
+    //    "modelCode" : "AppleTV5,3",
+    //    "identifier" : "CBB5E1ED-2172-446E-B4E7-F2B5823DBBA6",
+    //    "architecture" : "x86_64",
+    //    "modelName" : "Apple TV",
+    //    "name" : "Apple TV"
+    //  },
+    //  {
+    //    "simulator" : false,
+    //    "operatingSystemVersion" : "13.3 (17C54)",
+    //    "interface" : "usb",
+    //    "available" : true,
+    //    "platform" : "com.apple.platform.iphoneos",
+    //    "modelCode" : "iPhone8,1",
+    //    "identifier" : "d83d5bc53967baa0ee18626ba87b6254b2ab5418",
+    //    "architecture" : "arm64",
+    //    "modelName" : "iPhone 6s",
+    //    "name" : "iPhone"
+    //  },
+    //  {
+    //    "simulator" : true,
+    //    "operatingSystemVersion" : "6.1.1 (17S445)",
+    //    "available" : true,
+    //    "platform" : "com.apple.platform.watchsimulator",
+    //    "modelCode" : "Watch5,4",
+    //    "identifier" : "2D74FB11-88A0-44D0-B81E-C0C142B1C94A",
+    //    "architecture" : "i386",
+    //    "modelName" : "Apple Watch Series 5 - 44mm",
+    //    "name" : "Apple Watch Series 5 - 44mm"
+    //  },
+    // ...
+
+    final List<IOSDevice> devices = <IOSDevice>[];
+    for (final dynamic device in allAvailableDevices) {
+      if (device is! Map) {
+        continue;
+      }
+      final Map<String, dynamic> deviceProperties = device as Map<String, dynamic>;
+
+      // Only include iPhone, iPad, iPod, or other iOS devices.
+      if (!_isIPhoneOSDevice(deviceProperties)) {
+        continue;
+      }
+
+      final Map<String, dynamic> errorProperties = _errorProperties(deviceProperties);
+      if (errorProperties != null) {
+        final String errorMessage = _parseErrorMessage(errorProperties);
+        if (errorMessage.contains('not paired')) {
+          UsageEvent('device', 'ios-trust-failure').send();
+        }
+        _logger.printTrace(errorMessage);
+
+        final int code = _errorCode(errorProperties);
+
+        // Temporary error -10: iPhone is busy: Preparing debugger support for iPhone.
+        // Sometimes the app launch will fail on these devices until Xcode is done setting up the device.
+        // Other times this is a false positive and the app will successfully launch despite the error.
+        if (code != -10) {
+          continue;
+        }
```

Comment:
This is the changed logic.

Reply:
Does the new logic have a test?

Reply:
Yes, though GitHub decided to collapse xcode_test.dart. See 'ignores "Preparing debugger support for iPhone" error'.